### PR TITLE
Add `conj` as a special case and add promotion rule

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/function_registration.jl [hunk 1]
```diff
@@ -55,6 +55,7 @@ end
 # special cases
 Base.:^(x::Expression,y::T) where T <: Integer = Operation(Base.:^, Expression[x, y])
 Base.:^(x::Expression,y::T) where T <: Rational = Operation(Base.:^, Expression[x, y])
+Base.conj(x::Expression) = Operation(Base.conj, [x])
 Base.muladd(a::Expression, b::Expression, c::Expression) = a * b + c
```
Comment (ChrisRackauckas) [inline src/function_registration.jl]: why not make this actually a `muladd` call?

Comment (YingboMa) [reply src/function_registration.jl]: `muladd` has a promotion call, and that is not defined on `Expression`.

Comment (ChrisRackauckas) [reply src/function_registration.jl]: is there a way to bypass that?

Comment (YingboMa) [reply src/function_registration.jl]: Right, a better fix is to define a promotion rule.

**Pull request merged**